src/traverse/node-path.js

```diff
@@ -247,7 +300,11 @@ class NodePath {
       );
     }
 
-    return NodePath.registry.get(node);
+    let path = NodePath.registry.get(node);
+    if (parentPath != null) path.parentPath = parentPath;
+    if (prop       != null) path.property   = prop;
+    if (index      != null) path.index      = index;
```

Comment:
ok, it's your code -- but only under protest! :-)
seriously, it's certainly FAR more readable in tabular form, because you can directly see, that all lines are the same but for different names.